pyas2/tests/test_basic.py

```diff
@@ -534,6 +534,68 @@ def testEncryptSignMessageAsyncSignMdn(self, mock_request):
         mock_request.side_effect = RequestException()
         out_message.mdn.send_async_mdn()
 
+    @mock.patch("requests.post")
+    def testForceBinaryCanonicalization(self, mock_request):
+        """Test Permutation 15: Sender sends text data using binary canonicalization
+        and requests a synchronous receipt."""
+
+        partner = Partner.objects.create(
+            name="AS2 Server",
+            as2_name="as2server",
+            target_url="http://localhost:8080/pyas2/as2receive",
+            signature="sha1",
+            signature_cert=self.server_crt,
+            encryption=None,
+            encryption_cert=None,
+            mdn=True,
+            mdn_mode="SYNC",
+        )
+
+        receiver = Partner.objects.get(as2_name="as2client")
+        receiver.canonicalize_as_binary = True
+        receiver.save()
+
+        as2message = As2Message(
+            sender=self.organization.as2org, receiver=partner.as2partner
+        )
+
+        with mock.patch("pyas2lib.as2.canonicalize") as mock_canonicalize:
+            mock_canonicalize.side_effect = self.mock_canonicalize_function
+
+            as2message.build(
+                self.payload,
+                filename="testmessage.edi",
+                subject=partner.subject,
+                content_type=partner.content_type,
+            )
+
+            in_message, _ = Message.objects.create_from_as2message(
+                as2message=as2message, payload=self.payload, direction="OUT", status="P"
+            )
+
+        mock_request.side_effect = SendMessageMock(self.client)
+        in_message.send_message(
+            as2message.headers, self.mock_message_content(as2message)
+        )
+
+        # Check if message was processed successfully
+        out_message = Message.objects.get(
+            message_id=in_message.message_id, direction="IN"
+        )
+
+        receiver.canonicalize_as_binary = False
```

Comment:
why are we doing this here?

Reply:
The "receiver"'s canonicalize_as_binary is set to False by default in the setup. I do set it to True to test the functionality during the test and then set it back to the setup value. I probably should persist the initial value of the setup and the reset to that.